### ISSUE TITLE
Gitlab pipeline failes after automatic release

### ISSUE THREAD

**Issue by mrvonwyl** [semantic-release/semantic-release#1682]
## Current behavior

semantic-release is executed in Gitlab (self-managed) on the master branch. The releases are made and the new version is pushed on the master branch. This commit with the new version again triggers a pipeline in gitlab. And this second pipeline is what failes.
It seems like the semantic-release user does have rights to push on the branch (because of the access token) but not to run the pipeline.

![image](https://user-images.githubusercontent.com/2040951/98679465-60098000-2360-11eb-95f5-5a43798f9186.png)
semantic-release pushes the new version

![image](https://user-images.githubusercontent.com/2040951/98679293-14ef6d00-2360-11eb-8053-064745a64226.png)
it fails to execute the pipelines

This might well be a misconfiguration on the gitlab side. However, I was not able to figure anything out.

## Expected behavior

I expect that the pipelines triggered by the commit pushed by the semantic-release user completes successfully.

## Environment

- **semantic-release** version: 17.2.2
- CI environment: Gitlab
- Plugins used:
    "@semantic-release/changelog": "^5.0.1",
    "@semantic-release/commit-analyzer": "^8.0.1",
    "@semantic-release/git": "^9.0.0",
    "@semantic-release/npm": "^7.0.6",
    "@semantic-release/release-notes-generator": "^9.0.1",
- **semantic-release** configuration:
```
{
  "prepare": [
    "@semantic-release/changelog",
    "@semantic-release/npm",
    {
      "path": "@semantic-release/git",
      "assets": [
        "package.json",
        "package-lock.json",
        "CHANGELOG.md"
      ],
      "message": "Release: ${nextRelease.version}\n\n${nextRelease.notes}"
    }
  ],
  "release": {
    "branches": [
      "master"
    ]
  },
  "plugins": [
    [
      "@semantic-release/commit-analyzer",
      {
        "preset": "eslint"
      }
    ],
    "@semantic-release/release-notes-generator",
    "@semantic-release/npm",
    "@semantic-release/git",
    "@semantic-release/changelog"
  ]
}
```
- CI logs: 

First pipeline:
```
Running with gitlab-runner 13.4.1 (e95f89a0)
  on peax-gitlab-runner-2 aLAVS_63
Preparing the "docker" executor
Using Docker executor with image node:15 ...
Pulling docker image node:15 ...
Using docker image sha256:6c5fc80a4d2bf184190e25d80f21219e394fe348d17ac9758e3f50332d1d7d95 for node:15 with digest node@sha256:8ebfdeee577821f6c9caf672f3cb99fc2de39a9cb1d09a80a33d157c1452030f ...
Preparing environment
Running on runner-alavs63-project-265-concurrent-0 via a8bb61f9139c...
Getting source from Git repository
Fetching changes with git depth set to 50...
Reinitialized existing Git repository in /builds/peax3/code-conventions/.git/
Checking out 1c4b2e5c as master...
Removing node_modules/
Skipping Git submodules setup
Executing "step_script" stage of the job script
$ git config user.email "semantic-release@peax.ch"
$ git config user.name "Semantic Release"
$ npm ci
added 669 packages, and audited 1102 packages in 19s
1 package is looking for funding
  run `npm fund` for details
4 low severity vulnerabilities
To address all issues (including breaking changes), run:
  npm audit fix --force
Run `npm audit` for details.
$ npx semantic-release
[1:45:22 PM] [semantic-release] › ℹ  Running semantic-release version 17.2.2
[1:45:23 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[1:45:23 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/git"
[1:45:23 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/changelog"
[1:45:23 PM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[1:45:23 PM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[1:45:23 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/changelog"
[1:45:23 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[1:45:23 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/git"
[1:45:23 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[1:45:23 PM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/npm"
[1:45:23 PM] [semantic-release] › ✔  Run automated release from branch master on repository https://gitlab-ci-token:[secure]@gitlab.peax.ch/peax3/code-conventions.git
[1:45:24 PM] [semantic-release] › ✔  Allowed to push to the Git repository
[1:45:24 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/npm"
[1:45:24 PM] [semantic-release] [@semantic-release/npm] › ℹ  Verify authentication for registry https://registry.npmjs.org/
[1:45:24 PM] [semantic-release] [@semantic-release/npm] › ℹ  Reading npm config from /builds/peax3/code-conventions/.npmrc
mrvp
[1:45:24 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/npm"
[1:45:24 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/git"
[1:45:24 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/git"
[1:45:24 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/changelog"
[1:45:24 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/changelog"
[1:45:24 PM] [semantic-release] › ℹ  Found git tag v1.0.2 associated with version 1.0.2 on branch master
[1:45:24 PM] [semantic-release] › ℹ  Found 2 commits since last release
[1:45:24 PM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[1:45:24 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Merge branch 'develop'
[1:45:24 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[1:45:24 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Fix: correct after release pipeline
[1:45:24 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[1:45:24 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analysis of 2 commits complete: patch release
[1:45:24 PM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[1:45:24 PM] [semantic-release] › ℹ  The next release version is 1.0.3
[1:45:24 PM] [semantic-release] › ℹ  Start step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[1:45:24 PM] [semantic-release] › ✔  Completed step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[1:45:24 PM] [semantic-release] › ℹ  Start step "prepare" of plugin "@semantic-release/changelog"
[1:45:24 PM] [semantic-release] [@semantic-release/changelog] › ℹ  Update /builds/peax3/code-conventions/CHANGELOG.md
[1:45:24 PM] [semantic-release] › ✔  Completed step "prepare" of plugin "@semantic-release/changelog"
[1:45:24 PM] [semantic-release] › ℹ  Start step "prepare" of plugin "@semantic-release/npm"
[1:45:24 PM] [semantic-release] [@semantic-release/npm] › ℹ  Write version 1.0.3 to package.json in /builds/peax3/code-conventions
v1.0.3
[1:45:25 PM] [semantic-release] › ✔  Completed step "prepare" of plugin "@semantic-release/npm"
[1:45:25 PM] [semantic-release] › ℹ  Start step "prepare" of plugin "@semantic-release/git"
[1:45:25 PM] [semantic-release] [@semantic-release/git] › ℹ  Found 3 file(s) to commit
[1:45:26 PM] [semantic-release] [@semantic-release/git] › ℹ  Prepared Git release: v1.0.3
[1:45:26 PM] [semantic-release] › ✔  Completed step "prepare" of plugin "@semantic-release/git"
[1:45:26 PM] [semantic-release] › ℹ  Start step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[1:45:26 PM] [semantic-release] › ✔  Completed step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[1:45:27 PM] [semantic-release] › ✔  Created tag v1.0.3
[1:45:27 PM] [semantic-release] › ℹ  Start step "publish" of plugin "@semantic-release/npm"
[1:45:27 PM] [semantic-release] [@semantic-release/npm] › ℹ  Publishing version 1.0.3 to npm registry on dist-tag latest
npm notice 
npm notice package: @peax/code-conventions@1.0.3
npm notice === Tarball Contents === 
npm notice 262B  .editorconfig  
npm notice 60B   .huskyrc       
npm notice 160B  .lintstagedrc  
npm notice 153B  .prettierignore
npm notice 167B  .prettierrc    
npm notice 1.5kB .stylelintrc   
npm notice 1.6kB package.json   
npm notice 2.7kB tslint.json    
npm notice 309B  CHANGELOG.md   
npm notice 762B  README.md      
npm notice === Tarball Details === 
npm notice name:          @peax/code-conventions                  
npm notice version:       1.0.3                                   
npm notice package size:  3.1 kB                                  
npm notice unpacked size: 7.7 kB                                  
npm notice shasum:        9d63d64af1ca4dcf2797373248445e3807113224
npm notice integrity:     sha512-IN5gb+wSPY8WL[...]4eCG62plsfJSg==
npm notice total files:   10                                      
npm notice 
+ @peax/code-conventions@1.0.3
[1:45:32 PM] [semantic-release] [@semantic-release/npm] › ℹ  Published @peax/code-conventions@1.0.3 to dist-tag @latest on https://registry.npmjs.org/
[1:45:32 PM] [semantic-release] › ✔  Completed step "publish" of plugin "@semantic-release/npm"
[1:45:32 PM] [semantic-release] › ✔  Published release 1.0.3 on default channel
Cleaning up file based variables
Job succeeded
```
Second failing pipeline:
```
Running with gitlab-runner 13.4.1 (e95f89a0)
  on peax-gitlab-runner-1 nEVVF_dv
Preparing the "docker" executor
Using Docker executor with image node:15 ...
Pulling docker image node:15 ...
Using docker image sha256:6c5fc80a4d2bf184190e25d80f21219e394fe348d17ac9758e3f50332d1d7d95 for node:15 with digest node@sha256:8ebfdeee577821f6c9caf672f3cb99fc2de39a9cb1d09a80a33d157c1452030f ...
Preparing environment
Running on runner-nevvfdv-project-265-concurrent-0 via 100ea1ed6f96...
Getting source from Git repository
Fetching changes with git depth set to 50...
Reinitialized existing Git repository in /builds/peax3/code-conventions/.git/
remote: HTTP Basic: Access denied
fatal: Authentication failed for 'https://gitlab.peax.ch/peax3/code-conventions.git/'
Cleaning up file based variables
ERROR: Job failed: exit code 1
```

**Comment by ccll** [comment]
@mrvonwyl Stumbled on the same issue, after some digging it appears there are bugs in the GitLab Project Token handling, one need to use a GitLab Personal Token for GITLAB_TOKEN as for now.
The fix 'will be in the 13.7 release on the 22nd for self-managed' as stated here https://gitlab.com/gitlab-org/gitlab/-/issues/259665

**Comment by mrvonwyl** [comment]
@ccll Thank you for the hint. I checked and meanwhile we are on v3.7.3. It works as expected.
